code/modules/mob/living/simple_animal/bot/honkbot.dm

```diff
@@ -0,0 +1,398 @@
+				spam_flag = 1 // prevent spam
+		icon_state = "honkbot-e"
+		spawn(30) // keep flashing
+			icon_state = "honkbot[on]"
+		spawn(cooldowntimehorn)
+			spam_flag = 0
+
+/mob/living/simple_animal/bot/honkbot/proc/honk_attack(mob/living/carbon/C) // horn attack
+	if (!client) //check if a player is controlling
+		playsound(loc, honksound, 50, 1, -1)
+	else
+		if(spam_flag == 0)
+			playsound(loc, honksound, 50, 1, -1)
+			spam_flag = 1 // prevent spam
+	icon_state = "honkbot-c"
+	spawn(5)
+		icon_state = "honkbot[on]"
+	spawn(cooldowntimehorn)
+		spam_flag = 0
+
+/mob/living/simple_animal/bot/honkbot/proc/stun_attack(mob/living/carbon/C) // airhorn stun
+	if (!client)
+		playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HOOOOOOOOOOOOONK!!
+	else
+		if(spam_flag == 0)
+			playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HEEEEEEEEEEEENK!!
+	icon_state = "honkbot-c"
+	spawn(5)
+		icon_state = "honkbot[on]"
+	if(spam_flag == 0)
+		if(ishuman(C))
+			C.stuttering = 20
+			C.Jitter(50)
+			C.Knockdown(80)
+			var/mob/living/carbon/human/H = C
+			if (ckey != null) //don't alter AI behavior
+				spam_flag = 1
+			if (emagged <= 1) //HONK once, then leave
+				threatlevel = H.assess_threat(src)
+				threatlevel -= 6
+				//target = old_target
+			else // you really don't want to hit an emagged honkbot
+				threatlevel = 6 // will never let you go
+			spawn(cooldowntime)
+				spam_flag = 0
+		else
+			C.stuttering = 20
+			C.Knockdown(80)
+
+	add_logs(src,C,"honked")
+	spawn(cooldowntime)
+		spam_flag = 0
+	C.visible_message("<span class='danger'>[src] has honked [C]!</span>",\
+					"<span class='userdanger'>[src] has honked you!</span>")
+
+/mob/living/simple_animal/bot/honkbot/handle_automated_action()
+	if(!..())
+		return
+
+	switch(mode)
+
+		if(BOT_IDLE)		// idle
+
+			walk_to(src,0)
+			look_for_perp()
+			if(!mode && auto_patrol)
+				mode = BOT_START_PATROL
+
+		if(BOT_HUNT)
+
+			// if can't reach perp for long enough, go idle
+			if(frustration >= 8)
+				walk_to(src,0)
+				back_to_idle()
+				return
+
+			if(target)		// make sure target exists
+				if(Adjacent(target) && isturf(target.loc))
+
+					if(threatlevel <= 4)
+						honk_attack(target)
+					else
+						if(threatlevel >= 6)
+							spawn(0)
+								stun_attack(target)
+								anchored = 0
+								target_lastloc = target.loc
+					return
+
+				else	// not next to perp
+					var/turf/olddist = get_dist(src, target)
+					walk_to(src, target,1,4)
+					if((get_dist(src, target)) >= (olddist))
+						frustration++
+					else
+						frustration = 0
+			else
+				back_to_idle()
+
+
+		if(BOT_START_PATROL)
+			look_for_perp()
+			start_patrol()
+
+		if(BOT_PATROL)
+			look_for_perp()
+			bot_patrol()
+
+
+	return
+
+/mob/living/simple_animal/bot/honkbot/proc/back_to_idle()
+	anchored = 0
+	mode = BOT_IDLE
+	target = null
+	last_found = world.time
+	frustration = 0
```

Comment:
FALSE

Reply:
Frustration isn't a TRUE or FALSE variable. It builds up.  See BOT_HUNT.